script/hassfest/mypy_config.py

```diff
@@ -163,27 +144,9 @@ def _generate_and_validate_mypy_config(config: Config) -> str:  # noqa: C901
         else:
             strict_core_modules.append(module)
 
-    ignored_modules_set: set[str] = set(IGNORED_MODULES)
-    for module in strict_modules:
-        if (
-            not module.startswith("homeassistant.components.")
-            and module != "homeassistant.components"
-        ):
```

Comment:
Although slightly unrelated but this check is always false now because of how `strict_modules` is filled above.